Fixing gp default optimizer

Jimmy-INL opened the pull request:
--------
Pull Request Description
This PR proposes a quick fix for the name of the default optimizer in GP regressor.
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #1855
##### What are the significant changes in functionality due to this change request?
Fixes a defect.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

wangcj05 commented:
@Jimmy-INL You need to regold the test.

moosebuild commented:
Job [Test Ubuntu 18 PIP](https://civet.inl.gov/job/1079853/) on 3f4ec97 : invalidated by @Jimmy-INL

wangcj05 commented on tests/framework/ROM/SKLearn/tests:
```diff
@@ -145,11 +145,16 @@
+    [./csv]
+      type = OrderedCSV
+      output = 'data/outGP.csv'
+      rel_err = 5.0e06
```
should be 5.0E-06

wangcj05 commented on tests/framework/ROM/SKLearn/gp.xml:
```diff
@@ -28,35 +28,86 @@
       train,
-      resample
+      resample,
+      plot
```
please add revision information in the <TestInfo>

wangcj05 reviewed (changes requested):
I have some minor comments for you to consider

wangcj05 reviewed (approved):
PR is good.

wangcj05 commented on ravenframework/SupervisedLearning/ScikitLearn/GaussianProcess/GaussianProcessRegressor.py:
```diff
@@ -146,9 +146,9 @@ class cls.
+                                                 descr=r"""Per default, the 'fmin_l_bfgs_b' algorithm from
                                                  scipy.optimize.minimize is used. If None is passed, the kernel’s
-                                                 parameters are kept fixed. """, default='L-BGFS-B'))
+                                                 parameters are kept fixed. """, default='fmin_l_bfgs_b'))
```
I guess you need to regenerate the manual to reflect the change.

wangcj05 commented:
PR is good, but I think you need to regenerate the manual to reflect the changes. @Jimmy-INL

Jimmy-INL commented:
@wangcj05, I think I have fixed the manual issue. But maybe all other ROMs with the new API have it as well. If so, should I try to fix it here? Also, could you please take a look at what I have added in the docGenerator?

wangcj05 commented on doc/user_manual/generated/generateRomDoc.py:
```diff
@@ -775,18 +776,23 @@
 internalRom = ''
 sklROM = ''
 dnnRom = ''
+# from ravenframework import Optimizers
```
remove?

wangcj05 commented on doc/user_manual/generated/generateRomDoc.py:
```diff
@@ -775,18 +776,23 @@
 sklROM = ''
 dnnRom = ''
+# from ravenframework import Optimizers
+msg = ''
```
It seems 'msg' is not used.

wangcj05 commented on doc/user_manual/generated/generateRomDoc.py:
```diff
@@ -775,18 +776,23 @@
   try:
     if isinstance(obj(), ScikitLearnBase):
       sklROM += tex
+      sklROM += romDescr
```
why romDescr is after the input nodes description?

wangcj05 commented on ravenframework/SupervisedLearning/ScikitLearn/GaussianProcess/GaussianProcessRegressor.py:
```diff
@@ -129,26 +129,25 @@ class cls.
-                                                           values form a positive definite matrix. It can also be interpreted as the variance of additional Gaussian measurement noise on the training observations.""",
-                                                 default=1e-10))
+                                                           values form a positive definite matrix. It can also be interpreted as the variance of additional Gaussian measurement noise on the training observations. \default{1e-10}""",default=1e-10))
     specs.addSub(InputData.parameterInputFactory("n_restarts_optimizer", contentType=InputTypes.IntegerType,
                                                  descr=r"""The number of restarts of the optimizer for finding the kernel's parameters which maximize the log-marginal likelihood. The first run of the optimizer is performed
                                                            from the kernel's initial parameters, the remaining ones (if any) from thetas sampled log-uniform randomly from the space of allowed theta-values. If greater than
-                                                           0, all bounds must be finite. """, default=0))
+                                                           0, all bounds must be finite. \default{0}""", default=0))
     specs.addSub(InputData.parameterInputFactory("normalize_y", contentType=InputTypes.BoolType,
                                                  descr=r"""Whether the target values y are normalized, the mean and variance of the target values are set equal to 0 and 1 respectively. This is recommended for cases where zero-mean,
-                                                           unit-variance priors are used.""", default=False))
+                                                           unit-variance priors are used. \default{False}""",default=False ))
     specs.addSub(InputData.parameterInputFactory("random_state", contentType=InputTypes.IntegerType,
-                                              descr=r"""Seed for the internal random number generator""", default=None))
+                                              descr=r"""Seed for the internal random number generator. \default{None}""",default=None))
     specs.addSub(InputData.parameterInputFactory("optimizer", contentType=InputTypes.makeEnumType("optimizer", "optimizerType",['fmin_l_bfgs_b']),
-                                                 descr=r"""Per default, the 'L-BGFS-B' algorithm from
-                                                 scipy.optimize.minimize is used. If None is passed, the kernel’s
-                                                 parameters are kept fixed. """, default='L-BGFS-B'))
+                                                 descr=r"""Per default, the 'L-BFGS-b' algorithm from
+                                                 scipy.optimize.minimize is used. If None is passed, the kernel\’s
+                                                 parameters are kept fixed. \default{'fmin_l_bfgs_b'}""",default='fmin_l_bfgs_b'))
```
I think we should avoid to add \default directly in the 'descr'. In this case, we need to update all our roms. I would suggest that we could be able to use the default option to automatic generate \default tex msg, in this case, all our roms will be get updated instead of manual updates.

wangcj05 commented on ravenframework/SupervisedLearning/ScikitLearn/GaussianProcess/GaussianProcessRegressor.py:
```diff
@@ -168,6 +167,8 @@ def pickKernel(self, name):
       kernel = sklearn.gaussian_process.kernels.Exponentiation()
     elif name.lower() == 'matern':
       kernel = sklearn.gaussian_process.kernels.Matern()
+    elif name.lower() == 'pairwise':
+      kernel = sklearn.gaussian_process.kernels.PairwiseKernel()
```
I guess you can remove these lines.

wangcj05 reviewed (changes requested):
@Jimmy-INL I have a couples of comments for you to consider.

moosebuild commented:
Job [Mingw Test](https://civet.inl.gov/job/1106227/) on 71a21c1 : invalidated by @Jimmy-INL

wangcj05 commented:
@Jimmy-INL Do you need this PR for RAVEN workshop?

Jimmy-INL commented:
Yes
________________________________
From: Congjian Wang - INL ***@***.***>
Sent: Thursday, July 21, 2022 3:07:45 PM
To: idaholab/raven ***@***.***>
Cc: Mohammad G. Abdo ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [idaholab/raven] Fixing gp default optimizer (PR #1854)


@Jimmy-INL<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FJimmy-INL&data=05%7C01%7Cmohammad.abdo%40inl.gov%7C91dfbd659cd040b6ae7908da6b5d0f2f%7C4cf464b7869a42368da2a98566485554%7C0%7C0%7C637940344674268563%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=4HKzcNGF6Ij%2FH6WJ9oNJQLguNpDuyO3Lh50f7fhzQuY%3D&reserved=0> Do you need this PR for RAVEN workshop?

—
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fidaholab%2Fraven%2Fpull%2F1854%23issuecomment-1191936438&data=05%7C01%7Cmohammad.abdo%40inl.gov%7C91dfbd659cd040b6ae7908da6b5d0f2f%7C4cf464b7869a42368da2a98566485554%7C0%7C0%7C637940344674268563%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=fqj59a%2FqtzmDEj%2B3D2DEy0lVL5hCaoKdGxxlPrtD9Sw%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAMP5ECXTHBBGRPCT75JHO53VVG32DANCNFSM5YZFZY7A&data=05%7C01%7Cmohammad.abdo%40inl.gov%7C91dfbd659cd040b6ae7908da6b5d0f2f%7C4cf464b7869a42368da2a98566485554%7C0%7C0%7C637940344674268563%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=4N163AKnaMSyFLZibk4V5r%2Fgxik8T9O8RZGPW2J4s0I%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

moosebuild commented:
Job [Mingw Test](https://civet.inl.gov/job/1106239/) on 26c9530 : invalidated by @wangcj05

wangcj05 commented on ravenframework/utils/InputData.py:
```diff
@@ -661,14 +664,15 @@ def generateParamsLatex(cls, recDepth):
       default = '\\default{'+ str(info['default']) +'}' if info['default'] != 'no-default' else ""
+      default2 = 'default='+ str(info['default']) if info['default'] != 'no-default' else ""
       desc = wrapText(info['description'], indent=doDent(recDepth, 3))
-      if default:
+      if default or default2:
         msg += '\n{i}  \\item \\xmlAttr{{{n}}}: \\xmlDesc{{{t}, {r}}}, \n{d} {de}'.format(i=doDent(recDepth, 1),
                                                                                      n=name,
                                                                                      t=typ,
                                                                                      r=req,
                                                                                      d=desc,
-                                                                                     de=default)
+                                                                                     de=default2)
```
why default and default2 both used here? It seems to me you only need 1.

Jimmy-INL replied:
Some other places might still have the \default in the description. Should I just leave default2?

wangcj05 replied:
I may not understand the logic here, it seems to me 'default' and 'default2' are similar.

wangcj05 reviewed (changes requested):
I have one minor comment for the code, and another comment on the manual, it seems the manual is not updated to reflect the changes here.

moosebuild commented:
Job [Mingw Test](https://civet.inl.gov/job/1106239/) on 26c9530 : invalidated by @Jimmy-INL

wangcj05 commented:
@Jimmy-INL It seems civet is down. In order to merge your PR, you need to run the tests on your local machine, and report the results here. As well as the compilation of user manual.

Jimmy-INL commented:
These are the test results from my local machine since civit is down.
<img width="1440" alt="Screen Shot 2022-07-22 at 4 03 24 PM" src="https://user-images.githubusercontent.com/52417034/180574649-110771a4-19b9-4046-9a7e-54ad05dbee95.png">

wangcj05 commented on doc/user_manual/generated/generateRomDoc.py:
```diff
@@ -775,17 +776,20 @@
     continue
   obj = SupervisedLearning.factory.returnClass(name)
   specs = obj.getInputSpecification()
+  romDescr = wrapText(specs.description, ' ')
```
I checked the manual, it seems this information is already included. I think this line is not needed.

wangcj05 commented on doc/user_manual/generated/generateRomDoc.py:
```diff
@@ -775,17 +776,20 @@
   exampleTex = exampleFactory[name] if name in exampleFactory else ''
   try:
     if isinstance(obj(), ScikitLearnBase):
+      sklROM += romDescr
```
See my previous comments, this line is not needed.

wangcj05 reviewed (changes requested):
I have some comments for you to consider. After resolve my comments, please regenerate the rom tex file, and push to your branch. Thanks @Jimmy-INL

wangcj05 reviewed (approved):
PR looks good, thanks for your contribution. @Jimmy-INL

wangcj05 commented:
recent changes just to resolve reviewer's comment, just changes have been validated in local machine since civet test system is under maintenance now. Previous tests for commit "[26c9530](https://github.com/idaholab/raven/pull/1854/commits/26c9530dd7c7ff1a86622499916e02a78ab7893d)" is green. As CCB, I approve this PR.

wangcj05 commented:
checklist is satisfied. PR can be merged.
